### PR TITLE
[Improvement] Update estimator with api revision

### PR DESCRIPTION
**Pull request by gaoyang07**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

- Update ResourceEstimator with more init_cfgs.
- Add units prefix in measurement.
- Update UTs in estimator & runner_loops.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmrazor/models/task_modules/estimators/base_estimator.py [hunk 1]
```diff
@@ -12,44 +12,33 @@ class BaseEstimator(metaclass=ABCMeta):
...
-                 disabled_counters: List[str] = None,
-                 as_strings: bool = False,
-                 measure_inference: bool = False):
-        assert len(default_shape) in [3, 4, 5], \
-            f'Unsupported shape: {default_shape}'
-        self.default_shape = default_shape
+                 input_shape: Tuple = (1, 3, 224, 224),
+                 units: Dict = dict(),
+                 as_strings: bool = False):
+        assert len(input_shape) == 4, (
```
[the remark this answers is not in the document]
Comment (gaoyang07) [reply mmrazor/models/task_modules/estimators/base_estimator.py]: Done.

File: mmrazor/engine/runner/subnet_sampler_loop.py [hunk 2]
```diff
@@ -138,7 +140,8 @@ def __init__(self,
                  val_begin: int = 1,
                  val_interval: int = 1000,
                  score_key: str = 'accuracy/top1',
-                 flops_range: Optional[Tuple[float, float]] = (0., 330 * 1e6),
+                 flops_range: Optional[Tuple[float, float]] = (0., 330),
+                 resource_input_shape: Tuple = (1, 3, 224, 224),
```
[the remark this answers is not in the document]
Comment (gaoyang07) [reply mmrazor/engine/runner/subnet_sampler_loop.py:144]: Check commit [bbbb675](https://github.com/open-mmlab/mmrazor/pull/277/commits/bbbb675b251d95bc445d4d711e1126948c82c211) to assess current usage.

File: mmrazor/models/task_modules/estimators/resource_estimator.py [hunk 3]
```diff
@@ -1,38 +1,55 @@
...
-from .counters import (get_model_complexity_info, params_units_convert,
-                       repeat_measure_inference_speed)
+from .counters import get_model_flops_params, get_model_latency
 
 
 @TASK_UTILS.register_module()
 class ResourceEstimator(BaseEstimator):
     """Estimator for calculating the resources consume.
 
     Args:
```
Comment (humu789) [inline mmrazor/models/task_modules/estimators/resource_estimator.py:15]: docstring of args need to update

File: mmrazor/models/task_modules/estimators/counters/flops_params_counter.py [hunk 4]
```diff
@@ -1,26 +1,29 @@
...
-    print complexity information for each layer in a model. Supported layers
-    are listed as below:
+def get_model_flops_params(model,
+                           input_shape=(1, 3, 224, 224),
+                           spec_modules=[],
+                           disabled_counters=[],
+                           print_per_layer_stat=False,
+                           units=dict(flops='M', params='M'),
+                           as_strings=False,
+                           seperate_return: bool = False,
```
Comment (humu789) [inline mmrazor/models/task_modules/estimators/counters/flops_params_counter.py:19]: `seperate_return` need to add docstring

File: mmrazor/engine/runner/subnet_sampler_loop.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (humu789) [reply mmrazor/engine/runner/subnet_sampler_loop.py:144]: `resource_estimator_cfg` may be better, `evaluator` is confused easily with mmengine's. I will fix it in my commit

File: mmrazor/engine/runner/utils/check.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import copy
+from typing import Optional, Tuple
+
+import torch.nn as nn
+from mmdet.models.detectors import BaseDetector
```
Comment (sunnyxiaohu) [inline mmrazor/engine/runner/utils/check.py:6]: Try to use PlaceHolder for downstream repos.

Comment (gaoyang07) [reply mmrazor/engine/runner/utils/check.py]: Done.

**Pull request merged**